scripts/zones/Aht_Urhgan_Whitegate/npcs/Maudaal.lua

```diff
@@ -3,16 +3,33 @@
 --  NPC: Maudaal
 -- Standard Info NPC
 -----------------------------------
+local ID = require("scripts/zones/Aht_Urhgan_Whitegate/IDs")
 
 function onTrade(player,npc,trade)
 end
 
 function onTrigger(player,npc)
-    player:startEvent(240)
+    if player:getQuestStatus(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER) == QUEST_AVAILABLE then
+        player:startEvent(889)
+    elseif player:getQuestStatus(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER) == QUEST_ACCEPTED and player:getCharVar("princeandhopper") == 6 then
+        player:startEvent(890)
+    else
+        player:startEvent(240)
+    end
 end
-
 function onEventUpdate(player,csid,option)
 end
 
 function onEventFinish(player,csid,option)
-end
+    if (csid == 889 and option == 1) then
+        return
+    elseif (csid ==889 and option ==2) then
+        player:addQuest(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER)
+        player:setCharVar("princeandhopper",1)
+    elseif (csid ==890) then
+        player:addItem(16270)
+        player:messageSpecial(ID.text.ITEM_OBTAINED,16270)
```

Comment:
Probably wanna use `npcUtil.giveItem(player, 16270)` here as a condition for finishing the quest, does the same thing and checks inventory space, returns true/false.

Reply:
I somehow fucked this all up I need a masterclass on github for real.